gateway/handlers/forwarding_proxy.go

```diff
@@ -56,16 +62,19 @@ func MakeForwardingProxyHandler(proxy *types.HTTPClientReverseProxy, notifiers [
 
 		start := time.Now()
 
-		statusCode, err := forwardRequest(w, r, proxy.Client, baseURL, requestURL, proxy.Timeout)
+		statusCode, err := forwardRequest(w, r, proxy.Client, baseURL, requestURL, proxy.Timeout, writeRequestURI)
 
 		seconds := time.Since(start)
 		if err != nil {
 			log.Printf("error with upstream request to: %s, %s\n", requestURL, err.Error())
 		}
 
-		for _, notifier := range notifiers {
-			notifier.Notify(r.Method, requestURL, originalURL, statusCode, seconds)
-		}
+		defer func() {
```

Comment:
I may revert this, it was part of my testing, but by deferring it lets more throughput through the proxy since the client is not having to pay the latency of logging and writing to Prometheus which may involve acquiring an inode or mutex.

Reply:
I am still testing whether this affects throughput (I think it does) but have reverted for the PR.